testData/solutions/win32.json5

```diff
@@ -3724,6 +3724,85 @@
             }
         ]
     },
+    "com.microsoft.windows.desktopBackground": {
+        "name": "Windows desktop background personalization",
+        "contexts": {
+            "OS": [
+                {
+                    "id": "win32",
+                    "version": ">=5.0"
+                }
+            ]
+        },
+        "settingsHandlers": {
+            "configure": {
+                "type": "gpii.windows.desktopBackgroundSettingsHandler",
+                "liveness": "live",
+                "options": {},
+                "supportedSettings": {
+                    "BackgroundType": {
+                        "schema": {
+                            "title": "Desktop background payload",
+                            "description": "The type of background that user wants to select",
+                            "type": "string",
+                            "default": "Picture",
+                            "enum": [
+                                "SolidColor",
+                                "Picture",
+                                "Slideshow"
+                            ]
+                        }
+                    },
+                    "Input": {
+                        "schema": {
+                            "title": "Desktop background payload",
+                            "description": "The values that user want to use for that particular background type",
+                            "type": "Object",
+                            "default": {
+                                "Image": "‪C:\\Windows\\Web\\Wallpaper\\Windows\\img0.jpg",
+                                "Scaling": "Fill"
+                            },
+                            "enum": [
+                                // SolidColor 'BackgroundType' Input payload format
+                                {
+                                    "Color": "{ r: 0..0xff, g: 0..0xff, b: 0..0xff }"
```

Comment:
I don't believe that this use of enum is valid - @the-t-in-rtf , could you suggest a more effective way of encoding this constraint? Or would it be better to ontologise this with some common term holding a reusable transform which encodes RGB colours in some compact format such as the hex strings used here: https://github.com/the-grid/apidocs/issues/26

Reply:
The usage is correct, enum specifies literal values, including objects.  However, I dislike the practice of stuffing anything more complex than a string, number, or boolean in an enum array.  It seems like we're supporting schemas, when really, all we're doing is hard-coding a single example of an otherwise undocumented complex structure.  I think your idea about rgb colours as a string make some sense, as we are likely to need them in other contexts.  As long as no one is actually entering the deep value, having that be represented by a reusable transform seems acceptable, although I would love to see something (perhaps the thing interpreting the deep structure) providing a real schema to validate the more elaborate object structure.